torch/_inductor/scheduler.py

```diff
@@ -408,6 +408,12 @@ def decide_inplace_update(self) -> None:
         }
 
         ordered_reads = sorted(self.read_writes.reads, key=lambda x: x.name)
+        # NOTE remove V.graph.removed_operations once deps issue is fixed
+        inconsequential_nodes = (
+            (self.ancestors - {self.get_name()})
```

Comment:
i'm surprised a node is an ancestor of itself...

Reply:
I'm not sure if it's possible, do you know if the IR has been de-cycled when it gets to the scheduler? I added it defensibly in case there was a cycle, to prevent inplacing.

Reply:
Pretty sure a node should not be an ancestor of itself - you can add an assertion, run tests, then remove